substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/NativeImageClassLoaderSupport.java

```diff
@@ -845,6 +845,10 @@ private void handleClassFileName(URI container, Module module, String className)
                 clazz = imageClassLoader.forName(className, module);
             } catch (AssertionError error) {
                 VMError.shouldNotReachHere(error);
+            } catch (LinkageError le) {
+                synchronized (linkageErrors) {
+                    linkageErrors.put(className, le);
```

Comment:
Maybe we should make sure that the `LinkageError` does not reference another `Throwable` to prevent leakage?

Reply:
Wouldn't that risk losing the root cause?

To my understanding `LinkageError` may only reference another `Throwable` through the `cause` field.